superset/db_engine_specs/gsheets.py

```diff
@@ -35,11 +40,17 @@
 
 SYNTAX_ERROR_REGEX = re.compile('SQLError: near "(?P<server_error>.*?)": syntax error')
 
+ma_plugin = MarshmallowPlugin()
+
+
+class GSheetsParametersSchema(Schema):
+    catalog = fields.Dict()
+    query = fields.Dict(required=False)
+
 
 class GSheetsParametersType(TypedDict):
     credentials_info: Dict[str, Any]
-    query: Dict[str, Any]
-    table_catalog: Dict[str, str]
+    catalog: Dict[str, str]
```

Comment:
Wait, you asked me to change from `catalog` to `table_catalog` and now you're changing it back? 😄

Reply:
yeaa i reailzed the we needed it to match how we are saving it into engine spec :(, my bad on that